test/functional/ui/messages_spec.lua

```diff
@@ -1096,6 +1097,37 @@ vimComment     xxx match /\s"[^\-:.%#=*].*$/ms=s+1,lc=1  excludenl contains=@vim
       {4:Press ENTER or type command to continue}^                     |
     ]]}
   end)
+  it('msgfunc disables messages output', function()
```

Comment:
If feasible, it would be good to test the crash prevention as well. Something along the lines of:

*Note: this is pseudo-code which has not been ran*

```lua
it('msgfunc does not crash on recursion', function()
    source([[
    let g:ok = v:false
    function! Func(...) abort
      echomsg "Test crash prevention"
      let g:ok = v:true
    endfunction
    set msgfunc=Func
    ]])
    eq(0, eval('wait(2048, {->g:ok})'))
end)
```

Reply:
I have added the test.